src/executor/FaabricMain.cpp

```diff
@@ -17,6 +22,11 @@ namespace faabric::executor {
 
         conf.print();
 
+#if(FAASM_SGX)
+        // Check for SGX capability and create shared enclave
+        sgx::checkSgxSetup();
```

Comment:
The Faasm subclass of this class should invoke `checkSgxSetup` rather than have it here as an `extern`. If we do it here we are polluting the superclass with knowledge of its subclasses. This also brings knowledge of the Faasm repo into Faabric which we want to avoid.

There may need to be a `beforePoolSetup` hook that subclasses can create so that we provide the ability for them to do this at the right point.